File: jetbrains-plugin/src/main/kotlin/com/claudecodeplus/plugin/types/UiModels.kt
package com.claudecodeplus.plugin.types

import kotlinx.serialization.Serializable

/**
 * UI 模型类型定义（向后兼容）
 * 
 * 这些类型用于 JetBrains 插件中的 UI 层
 */

/**
 * AI 模型枚举
 */
enum class AiModel {
    SONNET,
    OPUS,
    HAIKU,
    OPUS_PLAN
}

/**
 * 权限模式枚举（插件 UI 用）
 */
enum class UiPermissionMode {
    DEFAULT,
    ACCEPT,
    BYPASS,
    PLAN
}

/**
 * 消息角色
 */
enum class MessageRole {
    USER,
    ASSISTANT,
    SYSTEM,
    ERROR
}

/**
 * 消息状态
 */
enum class MessageStatus {
    PENDING,
    STREAMING,
    COMPLETE,
    ERROR
}

/**
 * Token 使用信息（插件 UI 用，与 SDK 的 TokenUsage 不同）
 */
@Serializable
data class UiTokenUsage(
    val inputTokens: Int = 0,
    val outputTokens: Int = 0,
    val cacheCreationTokens: Int = 0,
    val cacheReadTokens: Int = 0,
    val totalTokens: Int = 0
)

/**
 * 会话对象（旧版，向后兼容）
 */
data class SessionObject(
    val id: String,
    val name: String,
    val createdAt: Long,
    val modelId: String?
)

/**
 * 增强消息（旧版，向后兼容）
 */
data class EnhancedMessage(
    val id: String,
    val role: MessageRole,
    val content: String,
    val timestamp: Long,
    val contexts: List<ContextReference> = emptyList(),
    val tokenUsage: UiTokenUsage? = null
)

/**
 * 旧版 ToolCall（向后兼容）
 * 现在是 ToolCallItem 的别名
 */
typealias ToolCall = ToolCallItem

/**
 * 旧版工具调用数据类（用于适配器层）
 */
data class LegacyToolCall(
    val name: String,
    val id: String,
    val status: String,
    val input: Map<String, Any?> = emptyMap(),
    val result: String? = null,
    val viewModel: Any? = null
)

File: vscode-extension/src/ide/types/uiModels.ts
/**
 * UI Models
 * 
 * UI model type definitions (backward compatible)
 * Translated from jetbrains-plugin/src/main/kotlin/com/asakii/plugin/types/UiModels.kt
 */

/**
 * AI Model enum
 */
export enum AiModel {
    SONNET = 'SONNET',
    OPUS = 'OPUS',
    HAIKU = 'HAIKU',
    OPUS_PLAN = 'OPUS_PLAN'
}

/**
 * UI Permission Mode enum
 */
export enum UiPermissionMode {
    DEFAULT = 'DEFAULT',
    ACCEPT = 'ACCEPT',
    BYPASS = 'BYPASS',
    PLAN = 'PLAN'
}

/**
 * Message role
 */
export enum MessageRole {
    USER = 'USER',
    ASSISTANT = 'ASSISTANT',
    SYSTEM = 'SYSTEM',
    ERROR = 'ERROR'
}

/**
 * Message status
 */
export enum MessageStatus {
    PENDING = 'PENDING',
    STREAMING = 'STREAMING',
    COMPLETE = 'COMPLETE',
    ERROR = 'ERROR'
}

/**
 * Token usage info (UI side, different from SDK's TokenUsage)
 */
export interface UiTokenUsage {
    inputTokens: number;
    outputTokens: number;
    cacheCreationTokens: number;
    cacheReadTokens: number;
    totalTokens: number;
}

export const defaultUiTokenUsage: UiTokenUsage = {
    inputTokens: 0,
    outputTokens: 0,
    cacheCreationTokens: 0,
    cacheReadTokens: 0,
    totalTokens: 0
};

/**
 * Session object (legacy, backward compatible)
 */
export interface SessionObject {
    id: string;
    name: string;
    createdAt: number;
    modelId?: string;
}

/**
 * Context reference
 */
export interface ContextReference {
    type: string;
    uri: string;
    displayType: string;
    path?: string;
    fullPath?: string;
    url?: string;
    title?: string;
    fileCount?: number;
    totalSize?: number;
    name?: string;
    mimeType?: string;
    base64Data?: string;
    size?: number;
}

/**
 * Enhanced message (legacy, backward compatible)
 */
export interface EnhancedMessage {
    id: string;
    role: MessageRole;
    content: string;
    timestamp: number;
    contexts?: ContextReference[];
    tokenUsage?: UiTokenUsage;
}

/**
 * Legacy tool call (for adapter layer)
 */
export interface LegacyToolCall {
    name: string;
    id: string;
    status: string;
    input?: Record<string, unknown>;
    result?: string;
    viewModel?: unknown;
}

/**
 * Create default session object
 */
export function createSessionObject(
    id: string,
    name: string,
    modelId?: string
): SessionObject {
    return {
        id,
        name,
        createdAt: Date.now(),
        modelId
    };
}

/**
 * Create enhanced message
 */
export function createEnhancedMessage(
    id: string,
    role: MessageRole,
    content: string,
    contexts?: ContextReference[],
    tokenUsage?: UiTokenUsage
): EnhancedMessage {
    return {
        id,
        role,
        content,
        timestamp: Date.now(),
        contexts,
        tokenUsage
    };
}
